provision/ansible/playbooks/roles/cassandra/files/disk_io_params.service

```
@@ -0,0 +1,11 @@
+[Unit]
+Description=Disk io parameters
+After=cloud-final.service
+Wants=cloud-final.service
+
+[Service]
+Type=simple
```

Probably `oneshot` is the right option?
https://github.com/scalar-labs/scalar-terraform/blob/74426c25952c5122c18c90b649e3fcd4d7e7c1ea/provision/ansible/playbooks/roles/cassandra/files/drivemount.service

```suggestion
Type=oneshot
RemainAfterExit=yes
```

---

Good point! 👍 
Fixed in 56cc0ad

---

I think `RemainAfterExit=yes` is also needed. Otherwise, it seems that the state will never become "active".

https://www.freedesktop.org/software/systemd/man/systemd.service.html
> Note that if this option is used without RemainAfterExit= the service will never enter "active" unit state, but directly transition from "activating" to "deactivating" or "dead" since no process is configured that shall run continuously.

---

I don't think it's necessary, Because it's not a resident process(`inactive`).

```
[centos@cassandra-1 ~]$ systemctl status disk_io_params
● disk_io_params.service - Disk io parameters
   Loaded: loaded (/etc/systemd/system/disk_io_params.service; enabled; vendor preset: disabled)
   Active: inactive (dead) since Tue 2021-03-23 08:37:04 UTC; 21s ago
  Process: 14607 ExecStart=/usr/local/bin/disk_io_params.sh (code=exited, status=0/SUCCESS)
 Main PID: 14607 (code=exited, status=0/SUCCESS)
```

```
RemainAfterExit=

Takes a boolean value that specifies whether the service shall be considered active even when all its processes exited. Defaults to no.
```

---

Right, but the `drivemount.service` specifies `RemainAfterExit=yes` with `Type=oneshot`. What do you think the difference is?

---

I need to update disk parameters by starting the service.
When set to `RemainAfterExit=yes` , it will not be possible to update after the first service startup. (difference with drivemount)

https://github.com/scalar-labs/scalar-terraform/pull/296/files#diff-610e26d306e2b7c67547e7b09f5681cfc6a10c59028ddad5c62b063af3a70522R14-R19

---

When `RemainAfterExit=yes` is set, isn't it possible to run the service again by `systemctl restart disk_io_params` as needed?

---

I've got the situation. Thanks!